equinor/witsml-explorer
# WE-479 CredentialsService `Authorization: Bearer ...` support

## Fixes
This pull request fixes `WE-479`

## Description
Handle use of `OAuth2` and `Bearer` tokens in `CredentialsService` in backend API

## How to test
`Basic` authentication should work as usual. This can be verified via swagger or frontend.

To test `OAuth2` flow with JWT token, enabling the use of system credentials from keyvault, the following are pre requisites (will be documented later in github)
* `mysettings.json` with config, for **backend** API to enable `OAuth2`
* `.env` settings updated for **frontend** to use `MSAL`
* App registration in Azure, along with app roles configured for admitted Azure AD users
* Witsml serverlist entries extended with properties for `securityscheme` and `roles`
* Key vault in Azure created, containing system credentials for witsml server.

After this has been configured, `swagger` can be used to get an accesstoken containing app-roles enabling access to designated witsml servers added to keyvault

**Log in via Swagger**

<img width="1015" alt="image" src="https://user-images.githubusercontent.com/1442452/189662077-f2f0e5f3-7fdc-4e8d-840b-b2bf83c6a410.png">

Use of API methods will now include `Authorization: Bearer ...` and backend API will access the witsml-server with system credentials fetched from Azure keyvault.


## Type of change

* New feature (non-breaking change which adds functionality)
* Enhancement of existing functionality
* This change requires a documentation update

## Impacted Areas in Application


## Checklist:

*Communication*
* [x] PR is related to an issue

*Code quality*
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments
Will be followed up by PRs extending the api client in frontend, and should be seen in relation to earlier PRs:
https://github.com/equinor/witsml-explorer/pull/1338
https://github.com/equinor/witsml-explorer/pull/1423
https://github.com/equinor/witsml-explorer/pull/1414
https://github.com/equinor/witsml-explorer/pull/1478
https://github.com/equinor/witsml-explorer/pull/1500
